Gateway/Request/InvoiceDataBuilder.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace Buckaroo\Magento2\Gateway\Request;
+
+use Magento\Payment\Gateway\Data\PaymentDataObjectInterface;
+use Magento\Payment\Gateway\Request\BuilderInterface;
+use Magento\Sales\Model\Order;
+
+class InvoiceDataBuilder implements BuilderInterface
+{
+    /**
+     * @var Order
+     */
+    private $order;
+
+    /**
+     * @var bool
+     */
+    private $isCustomInvoiceId = false;
+
+    /**
+     * @var string
+     */
+    private $invoiceId;
+
+
+    public function build(array $buildSubject)
+    {
+        if (!isset($buildSubject['payment'])
+            || !$buildSubject['payment'] instanceof PaymentDataObjectInterface
+        ) {
+            throw new \InvalidArgumentException('Payment data object should be provided');
+        }
+
+        $payment = $buildSubject['payment'];
+        $this->setOrder($payment->getOrder()->getOrder());
+
+        return [
+            'invoice' => $this->getInvoiceId(),
+            'order' => $this->getOrder()->getIncrementId()
+        ];
+    }
+
+    /**
+     * @return int
+     */
+    public function getInvoiceId()
+    {
+        $order = $this->getOrder();
```

Comment:
Will throw an error at the line: 
$this->setOrder($payment->getOrder()->getOrder());

Because getOrder must return Order object
app/code/Buckaroo/Magento2/Gateway/Data/Order/OrderAdapter.php:155